eventiq_asb/middlewares.py

```diff
@@ -146,5 +234,33 @@ async def _on_lock_renew_failure(
             exc_info=exc,
         )
 
-    async def after_broker_disconnect(self) -> None:
-        await self._renewer.close()
+    def stop_client(self, exc: threading.ExceptHookArgs) -> None:
+        asyncio.run(self.broker.client.close())
+        self.logger.error(
+            "Thread Error occurred in thread: %r. Error type: %s",
+            exc.thread,
+            exc.exc_type,
+            exc_info=exc.exc_value,
+        )
+
+    async def after_fail_message(
```

Comment:
not sure if putting msgs to ack_nack queue shouldn't be service responsibility, moreover we probably should send more actions then fail, ack, nack to worker thread (skip, retry, maybe even all actions related with message?)

Reply:
we could split logic from single ReceiverMiddleware to DLQReceiverMiddleware or something like this where we overwrite single method